### PR TITLE
Including per ROC histograms for Pixel Cluster Counting luminosity

### PR DESCRIPTION
**Pull request by duff-ae**
#### PR description:

The modules affected:
Calibration/LumiAlCaRecoProducers
DataFormats/Luminosity

Including extra loops to AlcaPCCEventProducer to iterate over clusters and sort them by corresponding ROC modules. Later the information is used to produce luminosity histograms (per bx) with per ROC granularity at HLT. This information could be used to identify the outliers and improve the stability and linearity of Pixel Cluster Counting luminosity (talk by B. Kronheim and C. Palmer https://indico.cern.ch/event/1358674/contributions/5725781/attachments/2775100/4836057/PCC_Active_Masking_Dec_19_2023.pdf)

The proposed future is validated following the official guideline http://cms-sw.github.io/PRWorkflow.html
- scram b runtests
- runTheMatrix.py -l limited -i all --ibeos:
    * Thu May 16 23:08:02 2024; exit: 0 0 0 47 46 44 35 18 1 1 1 1 1 1 tests passed, 0 0 1 0 0 0 0 0 0 0 0 failed
Also by running test reconstruction on raw data

Related PR:
https://github.com/cms-sw/cmssw/pull/29069

### CODE REVIEW

File: DataFormats/Luminosity/src/classes_def.xml [hunk 1]
```diff
@@ -46,12 +46,12 @@
   </class>
   <class name="std::vector<LumiSummary::L1>"/>
   <class name="std::vector<LumiSummary::HLT>"/>
-  <class name="reco::PixelClusterCounts" ClassVersion="3">
-   <version ClassVersion="3" checksum="1474294271"/>
```
Comment (perrotta) [inline DataFormats/Luminosity/src/classes_def.xml:50]: Do not remove previous class versions

File: DataFormats/Luminosity/src/classes_def.xml [hunk 2]
```diff
@@ -46,12 +46,12 @@
...
+  <class name="reco::PixelClusterCounts" ClassVersion="4">
+   <version ClassVersion="4" checksum="63275688"/>
   </class>
   <class name="edm::Wrapper<reco::PixelClusterCounts>"/>
-  <class name="reco::PixelClusterCountsInEvent" ClassVersion="3">
```
Comment (perrotta) [inline DataFormats/Luminosity/src/classes_def.xml:53]: Do not remove previous class versions

File: Calibration/LumiAlCaRecoProducers/plugins/AlcaPCCEventProducer.cc [hunk 3]
```diff
@@ -79,14 +85,25 @@ void AlcaPCCEventProducer::produce(edm::Event& iEvent, const edm::EventSetup& iS
...
+
+        /* generate the folling roc index that is going to map with ROC id as
+        8  9  10 11 12 13 14 15
+        0  1  2  3  4  5  6  7 */
+        int key = icol + irow * nROCcolumns;
+
+        // TODO: add roc threshold to config if(di.adc > fRocThreshold_) {
+        if (pix.adc > 0) {
+          thePCCob->incrementRoc(((detId << 7) + key), 1);
+        }
```
Comment (perrotta) [inline Calibration/LumiAlCaRecoProducers/plugins/AlcaPCCEventProducer.cc:103]:
```suggestion
        // TODO: add roc threshold to config if(di.adc > fRocThreshold_) {
        if (pix.adc > 0) {
          int irow = pix.x / rowsperroc; /* constant column direction is along x-axis */
          int icol = pix.y / colsperroc; /* constant row direction is along y-axis */
          /* generate the folling roc index that is going to map with ROC id as
          8  9  10 11 12 13 14 15
          0  1  2  3  4  5  6  7 */
          int key = icol + irow * nROCcolumns;
          thePCCob->incrementRoc(((detId << 7) + key), 1);
        }
```

File: DataFormats/Luminosity/src/classes_def.xml (hunk 1 above)
Comment (duff-ae) [reply DataFormats/Luminosity/src/classes_def.xml:50]: Dear Andrea, I resolved both cases. Commit was pushed

**Comment by perrotta** [comment]
please test

**Comment by mandrenguyen** [comment]
+1

**Comment by perrotta** [comment]
+alca

**Comment by rappoccio** [comment]
+1

**Comment by francescobrivio** [comment]
@duff-ae @cms-sw/alca-l2 do we need a backport for data-taking of this PR?

**Comment by francescobrivio** [comment]
@duff-ae @cms-sw/alca-l2 are you going to take care of creating the backport?

**Comment by perrotta** [comment]
Backport prepared in https://github.com/cms-sw/cmssw/pull/45054

**Comment by davidlange6** [comment]
This PR is the apparent cause of #45306 
(adding some GBs of RSS)

**Pull request merged**